### DISCUSSION TITLE
Sqlite3 growing very fast..

### DISCUSSION THREAD

**Discussion by emo44** [Ylianst/MeshCentral#5532] [General]
**Describe your issue**
Hello, since I changed to SQLITE3 the database has not stopped growing, in 10 days it has 40 megabytes, when before Neodb took up very little. This is normal? I only have 24-28 devices, what is the normal increase and is it possible that it is having some error and is not updating properly?

**Screenshots**
none

**Server Software (please complete the following information):**
 - OS: [Windows]
 - Virtualization: [e.g. Docker]
 - Network: [WAN]
 - Version: [1.1.0]
 - Node: [e.g. 14.17.1]

**Comment by PTR-inc** [comment]
For comparison, my sqlite db is a steady ~200Mb with around 40 devices of which around 20 permanently active. You could download a copy and check it with something like https://sqlitebrowser.org/ to see if there are abnormal tables/excessive entries.

As long as a db file is not absurdly large and you don't have issues (errors in de serverlog?), there is nothing to suspect. A db file always growns until it or you do maintance by compact/rebuild/vacuuming it. Also, different db's, different ways of storing data/logs.

**Comment by si458** [comment]
i noticed this too, when i was fixing the nedbtodb function the other month, i throught i was imaging it
but it just turned out while sqlite is better to see the database, structure, data, etc and more compataible than nedb,
it did infact use more space!

**Comment by silversword411** [comment]
Whitespace reservation in db fields?
How else can a zip of a 15GB database shrink to 1.2MB :D

**Comment by emo44** [comment]
ok, thank's for the information, perhaps we can share the results of the report with sqlite3_analyzer.exe in windows systems
https://www.sqlite.org/sqlanalyze.html

**Comment by raystek** [comment]
Is it safe to compact the SQLite database in meshcentral using the VACUUM command?

**Comment by PTR-inc** [comment]
This is somewhat fixed with https://github.com/Ylianst/MeshCentral/commit/e58d659fa94a74e6bbbef5198f85ec10e8a3993e
Old events weren't deleted. Added sqlite maintenance function to delete the old events and the backup does a 'vacuum into' to trim the backupfile. Sqlite by default doesn't shrink the active db file for performance reasons. It re-uses empty blocks. But if it is really big you could do this one time by simply making an autobackup, removing the .sqlite files and replacing the main sqlite file with the dump in the zip which is in itself just a sqlite db. With the maintenance function it should not grow that big anymore.
